ios/MusicControlManager.m

```diff
@@ -9,21 +9,56 @@
 @interface MusicControlManager ()
 
 @property (nonatomic, copy) NSString *artworkUrl;
+@property (nonatomic, strong) NSDictionary *mediaKeys;
 
 @end
 
+NSString * const MEDIA_PLAYBACK_RATE = @"playbackRate";
+
 @implementation MusicControlManager
 
 @synthesize bridge = _bridge;
 
 RCT_EXPORT_MODULE()
 
+- (NSDictionary *) mediaKeys {
+    if (_mediaKeys == nil) {
+        _mediaKeys = @{
+                @"album": MPMediaItemPropertyAlbumTitle,
+                @"trackCount": MPMediaItemPropertyAlbumTrackCount,
+                @"trackNumber": MPMediaItemPropertyAlbumTrackNumber,
+                @"artist": MPMediaItemPropertyArtist,
+                @"composer": MPMediaItemPropertyComposer,
+                @"discCount": MPMediaItemPropertyDiscCount,
+                @"discNumber": MPMediaItemPropertyDiscNumber,
+                @"genre": MPMediaItemPropertyGenre,
+                @"persistentID": MPMediaItemPropertyPersistentID,
+                @"duration": MPMediaItemPropertyPlaybackDuration,
+                @"title": MPMediaItemPropertyTitle,
+                @"elapsedPlaybackTime": MPNowPlayingInfoPropertyElapsedPlaybackTime,
+                @"playbackRate": MPNowPlayingInfoPropertyPlaybackRate,
```

Comment:
Android doesn't keep track of the position either. `getPosition` returns the position from when it was set, just like I stored the position somewhere and I were retrieving. If it returned the actual position, I didn't had to calculate the new position by myself.

Setting the speed to 0 will indeed be the same as `STATE_PAUSED`, as well as it would in Android. This is a programming problem with the app using this package, not ours.

It's fine if you want to require the app dev to set the elapsed time, feel free to do so. I might also change in my code, so I don't need to calculate it anymore. In Android, the elapsed time is only used and calculated by third party apps/devices such as smartwatches, that's why we can't retrieve in Android either.